lib/settingsChangeWatcher.ts

```diff
@@ -0,0 +1,37 @@
+import { ServerEvents } from '../app/models/server/raw';
+import { settings } from '../app/settings/server';
+import { IServerEventType } from '../definition/IServerEvent';
+import { SettingValue } from '../definition/ISetting';
+import { Logger } from '../app/logger/server';
+import { serverEvents } from './serverEvents';
+
+const logger = new Logger('SettingChangeWatcher');
+
+export const storeSettingChanged = async (settingId: string, targetValue: SettingValue, uid = 'system'): Promise<void> => {
+	const currentSetting = settings.get(settingId);
+	if (currentSetting === targetValue) {
+		logger.debug(`Cannot store change metadata for ${settingId}. Setting didn't change`);
+		return;
+	}
+	const event = {
+		t: IServerEventType.SETTING_MODIFIED,
+		ts: new Date(),
+		u: {
+			_id: uid,
```

Comment:
Since we're just storing the id of the user, so if this user is deleted from the server, then it means we won't be able to track back to the updater.
So, wanted to confirm if this is the intended behaviour? I feel this info might be required for audit purposes...Would it be okay if we also put the username or name within this user object?  .... maybe we can even create a setting for this to make it configurable